### DISCUSSION TITLE
Access to context object from module, display, interface, layout, panel extensions

### DISCUSSION THREAD

**Discussion by samdze** [directus/directus#13206] [Feature Requests]
Is there currently a way to access internal services, the database instance, the schema etc. (so, in short, the ApiExtensionContext object and maybe some way to get the accountability object) like we can do in hook and endpoint extensions, also in module, display and interface extensions?

Right now, I found you can only call `useApi` or `useStores` but they offer limited functionality and require you set up separate endpoints extensions to do anything other than simple data readings. Also, typescript definitions are a bit lacking, but that's another issue.

I understand that those types of extensions are primarily aimed at reading and displaying data, but having access to the context object would make creating powerful extensions much more easy.

This would apply especially for modules but also to allow complex queries in insights panels, e.g.
Or furthermore, display/interface extensions that provide buttons that execute some logic on internal services, the possibilities would be endless.

**Comment by rijkvanzanten** [comment]
The app itself is just a SPA that interfaces with the API through the endpoints, so you can't access any of the API internals from the app.

**Reply by samdze** [reply]
I see, makes sense.
The only thing that's not ideal is having to maintain several extensions and folders to provide the functionalities needed by the "core" extension you're making.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
